src/rust/ide/lib/view/src/graph_editor.rs

```diff
@@ -655,6 +658,22 @@ pub struct OptionalType(pub Option<ImString>);
 
 
 
+// =============================
+// === OptionalMethodPointer ===
+// =============================
+
+/// Information about target definition for node entering.
+///
+/// `None` means that the node cannot be entered.
+// TODO [mwu]
+//  As currently there is no good place to wrap Rc into a newtype that can be easily depended on
+//  both by `ide-view` and `ide` crates, we put this as-is. Refactoring should be considered in the
+//  future, once code organization and emerging patterns are more clear.
+#[derive(Clone,Debug,Default,Shrinkwrap)]
+pub struct OptionalMethodPointer(pub Option<Rc<enso_protocol::language_server::MethodPointer>>);
```

Comment:
Why Option is inside? `Rc` is for me implementation detail, but `Option` is a structural pattern telling that the field can be missing or not. I would definitely prefer to get `Option<MethodPointerSomething>` in frp rather than this, especially as there are special functions for handling options in frp, like for example `foo <= bar` - `foo` will be emitted only if `bar` was Some.